onnxruntime/core/providers/qnn/builder/qnn_def.cc

```diff
@@ -456,7 +456,7 @@ bool CreateTensorInQnnGraph(const QNN_INTERFACE_VER_TYPE& qnn_interface,
       return false;
     }
     // verify size expressed by the dims matches the raw tensor size
-    uint32_t qnn_tensor_size = CalcQnnTensorNumElems(qnn_tensor) * gsl::narrow_cast<uint32_t>(data_size);
+    uint32_t qnn_tensor_size = static_cast<uint32_t>(utils::GetQnnTensorDataSizeInBytes(qnn_tensor));
```

Comment:
does `qnn_tensor_size` need to be `uint32_t`?
```suggestion
    const auto qnn_tensor_size = utils::GetQnnTensorDataSizeInBytes(qnn_tensor);
```